api/v1beta1/common.go

```diff
@@ -83,3 +86,32 @@ func defaultIBMVPCMachineSpec(spec *IBMVPCMachineSpec) {
 		spec.Profile = "bx2-2x8"
 	}
 }
+
+func validateServiceEndpointURL(serviceEndpointURL string, region string) bool {
+	if strings.HasPrefix(serviceEndpointURL, "https://") || strings.HasPrefix(serviceEndpointURL, "http://") {
+		if strings.Contains(serviceEndpointURL, region) {
+			return true
+		}
+	}
+	return false
+}
+
+func validateIBMVPCClusterServiceEndpointURLValue(spec IBMVPCClusterSpec) bool {
+	if validateServiceEndpointURL(spec.ServiceEndpointURL, spec.Region) {
+		if strings.Contains(spec.ServiceEndpointURL, ".iaas.cloud.ibm.com/v1") {
+			return true
+		}
+	}
+	return false
+}
+
+func validateIBMPowerVSClusterServiceEndpointURLValue(spec IBMPowerVSClusterSpec) bool {
+	region, _, _ := resourcecontroller.GetResourceInstanceDetails(spec.ServiceInstanceID)
+	if validateServiceEndpointURL(spec.ServiceEndpointURL, region) {
+		if strings.Contains(spec.ServiceEndpointURL, ".power-iaas.cloud.ibm.com") {
+			return true
+		}
+		return true
+	}
+	return false
+}
```

Comment:
this flow may not work for the staging environment because staging URLs will be in the following format:

```
lon.power-iaas.test.cloud.ibm.com
iam.test.cloud.ibm.com/identity/token
```

Reply:
or user can have their own endpoints as well, lets avoid this validation itself.